packages/lib/apps/getEnabledApps.ts

```diff
@@ -2,12 +2,47 @@ import getApps from "@calcom/app-store/utils";
 import type { CredentialData } from "@calcom/app-store/utils";
 import { prisma } from "@calcom/prisma";
 
-const getEnabledApps = async (userCredentials: CredentialData[]) => {
+/**
+ *
+ * @param credentials - Can be user or team credentials
+ * @param filterOnCredentials - Only include apps where credentials are present
+ * @returns A list of enabled app metadata & credentials tied to them
+ */
+const getEnabledApps = async (credentials: CredentialData[], filterOnCredentials?: boolean) => {
+  const filterOnIds = {
+    credentials: {
+      some: {
+        OR: [],
+      },
+    },
+  };
+  if (filterOnCredentials) {
+    const userIds: number[] = [],
+      teamIds: number[] = [];
+
+    for (const credential of credentials) {
+      if (credential.userId) userIds.push(credential.userId);
+      if (credential.teamId) teamIds.push(credential.teamId);
+    }
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //@ts-ignore
```

Comment:
Since the inital object `OR` is an empty array it is of type `never`. Although the prisma `where` object expects `AppWhereInput`